python/triton/runtime/autotuner.py

```diff
@@ -102,14 +103,18 @@ def kernel_call():
             if config.pre_hook:
                 config.pre_hook(full_nargs)
             self.pre_hook(args)
-            self.fn.run(
-                *args,
-                num_warps=config.num_warps,
-                num_stages=config.num_stages,
-                num_ctas=config.num_ctas,
-                **current,
-            )
-            self.post_hook(args)
+            try:
+                self.fn.run(
+                    *args,
+                    num_warps=config.num_warps,
+                    num_stages=config.num_stages,
+                    num_ctas=config.num_ctas,
+                    **current,
+                )
+            except OutOfResources as e:
+                raise e
+            finally:
+                self.post_hook(args, from_exception=True)
```

Comment:
Presumably we should pass from_exception=False in the other place where we call post_hook?

Reply:
Ah, I think there's a logical problem. Thanks for point it out. Only when we have an exception, `from_exception=True`.